src/services/HmrcService.ts

```diff
@@ -0,0 +1,76 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Constants } from "../utils/Constants";
+import axios from "axios";
+import { AppError } from "../utils/AppError";
+import { getParameter } from "../utils/Config";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { HmrcVerifyResponse } from "../models/IHmrcResponse";
+
+export class HmrcService {
+	readonly logger: Logger;
+
+	private static instance: HmrcService;
+
+    readonly HMRC_BASE_URL: string;
+
+    readonly HMRC_CLIENT_ID: string | undefined;
+
+    readonly HMRC_CLIENT_SECRET: string | undefined;
+
+    constructor(logger: Logger, HMRC_BASE_URL: string, HMRC_CLIENT_ID?: string, HMRC_CLIENT_SECRET?: string) {
+    	this.logger = logger;
+    	this.HMRC_BASE_URL = HMRC_BASE_URL;
+    	this.HMRC_CLIENT_ID = HMRC_CLIENT_ID;
+    	this.HMRC_CLIENT_SECRET = HMRC_CLIENT_SECRET;
+    }
+
+    static getInstance(
+    	logger: Logger,
+    	HMRC_BASE_URL: string,
+    	HMRC_CLIENT_ID?: string,
+    	HMRC_CLIENT_SECRET?: string,
+    ): HmrcService {
+    	if (!HmrcService.instance) {
+    		HmrcService.instance = new HmrcService(logger, HMRC_BASE_URL, HMRC_CLIENT_ID, HMRC_CLIENT_SECRET);
+    	}
+    	return HmrcService.instance;
+    }
+
+    // eslint-disable-next-line max-lines-per-function
+    async verify(
+    	{ accountNumber, sortCode, name }: { accountNumber: string; sortCode: string; name: string }, hmrcTokenSsmPath: string,
+    ): Promise<HmrcVerifyResponse> {
+    	const token = await getParameter(hmrcTokenSsmPath);
+    	const params = {
+    		account: { accountNumber, sortCode },
+    		subject: { name },
+    	};
+    	const headers = {
+    		"User-Agent": Constants.HMRC_USER_AGENT,
+    		"Authorization": `Bearer ${token}`,
+    	};
+
+    	try {
+    		this.logger.info("Sending COP verify request to HMRC");
+    		const { data }: { data: HmrcVerifyResponse } = await axios.post(`${this.HMRC_BASE_URL}/${Constants.HMRC_VERIFY_ENDPOINT_PATH}`, params, { headers });
+
+    		this.logger.debug({
```

Comment:
Have we confirmed none of these will be PII?

Reply:
@mudassar-jafer has confirmed, think I need to check with anyone else?

Reply:
All good :)